k-distribution/src/main/scripts/bin/kx

```diff
@@ -449,12 +463,13 @@ else
       unescaped=$(</dev/stdin)
       stdinText=$(awk 'BEGIN {for(n=0;n<256;n++)ord[sprintf("%c",n)]=n} {text = text $0 ORS} END {split(text, chars, ""); for (i=1; i<=length(text); i++) { if (chars[i] == "\"") { printf("%s", "\\\"") } else if (chars[i] == "\\") { printf("%s", "\\\\") } else if (chars[i] == "\n") { printf("%s", "\\n") } else if (chars[i] == "\t") { printf("%s", "\\t") } else if (chars[i] == "\r") { printf("%s", "\\r") } else if (chars[i] == "\f") { printf("%s", "\\f") } else if (ord[chars[i]] >= 32 && ord[chars[i]] < 127) { printf("%s", chars[i]) } else { printf("\\x%02x", ord[chars[i]]) } } }' <<< "$unescaped")
     fi
-    stdinFile="$(mktemp tmp.stdin.XXXXXXXXXX)"
+    stdinFile="$(mktemp ${tempDir}/tmp.stdin.XXXXXXXXXX)"
     tempFiles+=("$stdinFile")
     printf '\dv{SortString{}}(\"%s\")\n' "$stdinText" > "$stdinFile"
-    configVars="$configVars -c STDIN $stdinFile String korefile"
+    configVars="$configVars -c STDIN $(basename $stdinFile) String korefile"
   fi
-  execute llvm-krun $configVars -d "$kompiledDir" $flags --dry-run -o "$input_file"
+  # llvm-krun creates temp files in the cwd so execute it in $tempDir instead
+  (cd $tempDir; execute llvm-krun $configVars -d "../$kompiledDir" $flags --dry-run -o "$(basename $input_file)")
```

Comment:
This is something which I don't understand.
Why is this call needed?

Reply:
Uh, it creates the initial configuration that gets passed to the interpreter? It's a very important step.